src/core/edgedb/seeds.run.ts

```diff
@@ -0,0 +1,98 @@
+import { many, MaybeAsync } from '@seedcompany/common';
+import { BaseContext, Command, runExit } from 'clipanion';
+import { Client, createClient, Executor } from 'edgedb';
+import { glob } from 'glob';
+import fs from 'node:fs/promises';
+import { inspect } from 'util';
+import { e } from './reexports';
+
+type Query = string | QBQuery;
+interface QBQuery {
+  run: (client: Executor) => Promise<any>;
+}
+
+export type SeedFn = (
+  params: SeedParams,
+) => MaybeAsync<void | Query | Iterable<Query> | AsyncIterable<Query>>;
+
+interface SeedParams {
+  e: typeof e;
+  runAndPrint: (query: Query) => Promise<void>;
+  db: Client;
+  print: (something: unknown) => void;
+  context: BaseContext;
+}
+
+const db = createClient().withConfig({
+  // eslint-disable-next-line @typescript-eslint/naming-convention
+  allow_user_specified_id: true,
+  // eslint-disable-next-line @typescript-eslint/naming-convention
+  apply_access_policies: false,
+});
+
+class SeedCommand extends Command {
+  async execute() {
+    let code = 0;
+    const files = await glob('./dbschema/seeds/**/*.{edgeql,ts}');
+    files.sort((a, b) => a.localeCompare(b));
+
+    const colors = this.context.colorDepth > 1;
+    const printResult = (result: unknown) => {
+      for (const row of many(result)) {
+        this.context.stdout.write(inspect(row, { colors }) + '\n');
+      }
+    };
+
+    const runAndPrint = async (query: Query) => {
+      try {
+        const rows =
+          typeof query === 'string'
+            ? await db.query(query)
+            : await query.run(db);
+        printResult(rows);
+      } catch (e) {
+        this.context.stderr.write((e as Error).message + '\n');
+        code = 1;
+      }
+    };
+    const params: SeedParams = {
+      db,
+      context: this.context,
+      print: printResult,
+      runAndPrint,
+      e,
+    };
+
+    for (const file of files) {
+      if (file.endsWith('.edgeql')) {
+        const query = await fs.readFile(file, 'utf-8');
+        await runAndPrint(query);
+      } else {
+        const script = await import('../../../' + file);
+        const queries = await (script.default as SeedFn)(params);
+        if (!queries) {
+          continue;
+        }
```

Comment:
It's a nice approach, but 72 just looks weird, casting and doing a call on what was casted. I'm sure this'll be a low maintenance part of our code, so I'm ok leaving it as is for now, but that like gives me the "icks".

Reply:
I think I didn't like my variable name for this fn & typecast so my immediate next thought was just to inline it lol